### PR TITLE
Feature/query datetime subset fix

### PR DESCRIPTION
**Pull request by rmclaren**
## Description

This pull request does several things:

- Filters the data down to only the data for the subsets that are included in the queries as in the example in the following file: test/testinput/bufr_specific_subsets_by_query.yaml.
- Adds a new section in exports called subsets. (see the following: test/testinput/bufr_specifying_subsets.yaml)
- Fixed some issues in the DatetimeVaraibles with regard to the interpretation of missing values.
- Includes some refactoring. The QuerySet now uses the QueryParser and queries are parsed into Query objects. The queryset can now tell you if a subset is one that needs to be processed or not.

### Issue(s) addressed

- https://github.com/JCSDA-internal/ioda-converters/issues/1047

## Acceptance Criteria (Definition of Done)

- YAML file supports new exports field called subsets which can be used to limit the subsets that will be processed.
-  Only subsets that are specified in queries will be processed (* means everything).
- Fixes bugs with DatetimeVariable object.

## Dependencies

- https://github.com/JCSDA-internal/ioda-converters/pull/1055

### CODE REVIEW

File: src/bufr/DataObject.h [hunk 1]
```diff
@@ -110,9 +110,9 @@ namespace Ingester
         /// \return Float data.
         virtual float getAsFloat(const Location& loc) const = 0;
 
-        /// \brief Get the data at the index as an float.
+        /// \brief Get the data at the index as an int.
         /// \return Float data.
-        virtual float getAsFloat(size_t idx) const = 0;
+        virtual int getAsInt(size_t idx) const = 0;
```
Comment (nicholasesposito) [inline src/bufr/DataObject.h:115]: Ron, I am using the getAsFloat function in my feature/query_cxx_types_NE_aircraftAltitude branch. I think we should keep both

Comment (rmclaren) [reply src/bufr/DataObject.h:115]: Actually I really need to get rid of both as using this function is really dangerous if you are not careful... If you depend on any values in repeated sequences (or that could be in repeated sequences) you should not go near these methods. I'm gonna have to think about this a little I think.

Comment (nicholasesposito) [reply src/bufr/DataObject.h:115]: OK, I'll rework my code to not use it.

**Review by nicholasesposito — APPROVED** [review]
Lots of good changes in this update, especially regarding the determination of array sizes. I will definitely use this to fix up my own aircraftAltitude. Thanks for doing this.

File: src/bufr/DataObject.h (hunk 1 above)
Comment (rmclaren) [reply src/bufr/DataObject.h:115]: Ok, I added it back. You need to validate the input data and make sure it all comes from the same place (same place I the tree).

Comment (rmclaren) [reply src/bufr/DataObject.h:115]: This will need to be revisited in the future.

**Pull request merged**